.ebextensions/nginx.config

```diff
@@ -0,0 +1,232 @@
+          text/vtt                              vtt;
+          text/x-component                      htc;
+
+      }
+
+  "/etc/nginx/extras/directive-only/extra-security.conf":
+    mode: "000644"
+    owner: root
+    group: root
+    content: |
+      # The X-Frame-Options header indicates whether a browser should be allowed
+      # to render a page within a frame or iframe.
+      add_header X-Frame-Options SAMEORIGIN always;
+
+      # MIME type sniffing security protection
+      #	There are very few edge cases where you wouldn't want this enabled.
+      add_header X-Content-Type-Options nosniff always;
+
+      # The X-XSS-Protection header is used by Internet Explorer version 8+
+      # The header instructs IE to enable its inbuilt anti-cross-site scripting filter.
+      add_header X-XSS-Protection "1; mode=block" always;
+
+      # with Content Security Policy (CSP) enabled (and a browser that supports it (http://caniuse.com/#feat=contentsecuritypolicy),
+      # you can tell the browser that it can only download content from the domains you explicitly allow
+      # CSP can be quite difficult to configure, and cause real issues if you get it wrong
+      # There is website that helps you generate a policy here http://cspisawesome.com/
+      add_header Content-Security-Policy "default-src 'self'; script-src 'self' 'unsafe-inline' https://dap.digitalgov.gov https://www.google-analytics.com; img-src 'self' *.tile.osm.org; frame-ancestors 'none';" always;
+
+  "/etc/nginx/extras/directive-only/x-ua-compatible.conf":
+    mode: "000644"
+    owner: root
+    group: root
+    content: |
+      # Force the latest IE version
+      add_header X-UA-Compatible "IE=Edge";
+
+  "/etc/nginx/extras/location/expires.conf":
+    mode: "000644"
+    owner: root
+    group: root
+    content: |
+      # Expire rules for static content
+
+      # No default expire rule. This config mirrors that of apache as outlined in the
+      # html5-boilerplate .htaccess file. However, nginx applies rules by location,
+      # the apache rules are defined by type. A consequence of this difference is that
+      # if you use no file extension in the url and serve html, with apache you get an
+      # expire time of 0s, with nginx you'd get an expire header of one month in the
+      # future (if the default expire rule is 1 month). Therefore, do not use a
+      # default expire rule with nginx unless your site is completely static
+
+      # cache.appcache, your document html and data
+      location ~* \.(?:manifest|appcache|html?|xml|json)$ {
+        add_header Cache-Control "max-age=0";
+      }
+
+      # Feed
+      location ~* \.(?:rss|atom)$ {
+        add_header Cache-Control "max-age=3600";
+      }
+
+      # Media: images, icons, video, audio, HTC
+      location ~* \.(?:jpg|jpeg|gif|png|ico|cur|gz|svg|webp|mp4|ogg|ogv|webm|htc)$ {
+        access_log off;
+        add_header Cache-Control "max-age=2592000";
```

Comment:
does this mean media is cached for 30 days?
if i change an icon and keep the name the same will the old one be cached?

Reply:
> does this mean media is cached for 30 days?

Correct!

> if i change an icon and keep the name the same will the old one be cached?

Rails handles cache-busting by generating fingerprinted URLs on production for media files that have changed since the previous deploy.